src/main/java/org/truffleruby/core/module/ModuleOperations.java

```diff
@@ -395,71 +394,91 @@ public static ConstantLookupResult lookupConstantWithInherit(RubyContext context
         return definedMethods;
     }
 
-    @TruffleBoundary
     public static MethodLookupResult lookupMethodCached(DynamicObject module, String name,
             DeclarationContext declarationContext) {
+        return lookupMethodCached(module, null, name, declarationContext);
+    }
+
+    @TruffleBoundary
+    private static MethodLookupResult lookupMethodCached(DynamicObject module, DynamicObject lookupTo, String name,
+            DeclarationContext declarationContext) {
         final ArrayList<Assumption> assumptions = new ArrayList<>();
 
-        // Look in ancestors
```

Comment:
Could you keep this comment? It helps to follow the logic in the various lookup methods without needing to check the details below everytime.